src/registrar/views/portfolios.py

```diff
@@ -216,6 +220,68 @@ def get(self, request, pk):
             },
         )
 
+    def post(self, request, pk):
+        """
+        Handles adding and removing domains for a portfolio member.
+        """
+        added_domains = request.POST.get("added_domains")
+        removed_domains = request.POST.get("removed_domains")
+        portfolio_permission = get_object_or_404(UserPortfolioPermission, pk=pk)
+        member = portfolio_permission.user
+
+        added_domain_ids = self._parse_domain_ids(added_domains, "added domains")
+        if added_domain_ids is None:
+            return redirect(reverse("member-domains", kwargs={"pk": pk}))
+
+        removed_domain_ids = self._parse_domain_ids(removed_domains, "removed domains")
+        if removed_domain_ids is None:
+            return redirect(reverse("member-domains", kwargs={"pk": pk}))
+
+        if added_domain_ids or removed_domain_ids:
+            try:
+                self._process_added_domains(added_domain_ids, member)
+                self._process_removed_domains(removed_domain_ids, member)
+                messages.success(request, "The domain assignment changes have been saved.")
+                return redirect(reverse("member-domains", kwargs={"pk": pk}))
+            except IntegrityError:
+                messages.error(request, "A database error occurred while saving changes.")
+                return redirect(reverse("member-domains-edit", kwargs={"pk": pk}))
+            except Exception as e:
+                messages.error(request, f"An unexpected error occurred: {str(e)}")
```

Comment:
**change before merging** same comments in this block that I mention below, log errors and add language so that a user emails help@get.gov if the error persists.